src/main/java/edu/cornell/kfs/sys/dataaccess/impl/PreferencesDaoMongo.java

```diff
@@ -0,0 +1,204 @@
+package edu.cornell.kfs.sys.dataaccess.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import org.apache.commons.collections.CollectionUtils;
+import org.kuali.kfs.sys.dataaccess.PreferencesDao;
+import org.springframework.data.mongodb.core.index.IndexDefinition;
+import org.springframework.data.mongodb.core.index.IndexInfo;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PreferencesDaoMongo implements PreferencesDao {
```

Comment:
Is the abstraction of mongoTemplate behind the documentstoreClient the only difference between this and the default KFS PreferencesDaoMongo? If so, and if you expect getClient() to always return the same object, then perhaps this could be just a subclass of the existing DAO, and relies on a setter or other configuration to populate the mongoTemplate reference from the documentstoreClient. That is, unless this abstraction needs to be maintained in case getClient() can return various instances.